drivers/metal/metal_objects.mm

```diff
@@ -1417,9 +1426,24 @@
 	}
 }
 
+uint32_t MDUniformSet::calculate_dynamic_state_mask() const {
+	uint32_t mask = 0u;
+	uint32_t shift = 0u;
+	for (const MetalBufferDynamicInfo *dynamic_buffer : dynamic_buffers) {
+		DEV_ASSERT(dynamic_buffer->frame_idx < 16u);
+		mask |= dynamic_buffer->frame_idx << shift;
+		shift += 4u;
```

Comment:
Given the range of frame index is [1, 3] (2-4 four frames), these values could be reduced

```suggestion
		DEV_ASSERT(dynamic_buffer->frame_idx < 4u);
		mask |= dynamic_buffer->frame_idx << shift;
		shift += 2u;
```